x-pack/plugins/security/server/role_mappings/get_enabled_role_mappings_features.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IClusterClient, Logger } from 'src/core/server';
+
+interface Deps {
+  logger: Logger;
+  clusterClient: IClusterClient;
+}
+
+interface NodeSettingsResponse {
+  nodes: {
+    [nodeId: string]: {
+      settings: {
+        script: {
+          allowed_types?: string[];
+          allowed_contexts?: string[];
+        };
+      };
+    };
+  };
+}
+
+interface XPackUsageResponse {
+  security: {
+    realms: {
+      [realmName: string]: {
+        available: boolean;
+        enabled: boolean;
+      };
+    };
+  };
+}
+
+const INCOMPATIBLE_REALMS = ['file', 'native'];
+
+export const getEnabledRoleMappingsFeatures = async ({ clusterClient, logger }: Deps) => {
+  logger.debug(`Retrieving role mappings features`);
+
+  const nodeScriptSettingsPromise: Promise<NodeSettingsResponse> = clusterClient
+    .callAsInternalUser('transport.request', {
+      method: 'GET',
+      path: '/_nodes/settings?filter_path=nodes.*.settings.script',
+    })
+    .catch(error => {
+      logger.error(`Error retrieving node settings for role mappings: ${error}`);
+      return {};
+    });
+
+  const xpackUsagePromise: Promise<XPackUsageResponse> = clusterClient
+    .callAsInternalUser('transport.request', {
+      method: 'GET',
+      path: '/_xpack/usage',
```

Comment:
I didn’t do this originally because it used to be called the shield plugin, and this route isn’t related to security. Are we ok mixing the responsibilities of this ES client plugin?

Reply:
That's a great question that likely warrants further discussion. Our usage of these elasticsearch client plugins needs some rework, and after further investigation there seems to be a rather prolific use of `transport.request` to get around the lack of am "x-pack plugin". Let's not blow the scope up on this PR and we can just leave it the way it is now. Do you mind adding a comment to it about user-input being used to construct the URL?